### PR TITLE
.github: check connectivity of external links via GitHub action

### PR DESCRIPTION
**Pull request by yikeke**
### What is changed, added or deleted? (Required)

A migration from https://github.com/pingcap/docs-dm/pull/353

**Note** this external link check isn't required to pass, in order to prevent false alarms from blocking PR merge.

The current regex exclusions is: 

`http://172.*|https://github.com/.*/pull/.*|https://github.com/.*/issues/.*|https://github.com/[^.\n]*$|http://127.0.0.1.*|http://localhost.*|http://.*:.*|https://static.googleusercontent.com/media/research.google.com/.*|.*.md`

Meaning **the check will skip checking the URLs matching the above patterns**.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Have version specific changes 
- [ ] Might cause conflicts

### CODE REVIEW

**Comment by yikeke** [comment]
除了[这个问题](https://github.com/pingcap/docs-tidb-operator/pull/694#issuecomment-689353940)，又发现了个问题：目前报的有些链接有些是 false alarms，比如 `http://127.0.0.1:2379/dashboard/#/cluster_info/instance`、`https://github.com/$user/docs-cn` 之类的链接，这些都是已知的、特意用 <> 扩起来的链接。liche 这个工具支持 exclude 掉包裹在 <> 的链接吗？还是只能用 regex 挨个过滤掉这些特定的链接呢？ @csuzhangxc

**Comment by csuzhangxc** [comment]
> 除了[这个问题](https://github.com/pingcap/docs-tidb-operator/pull/694#issuecomment-689353940)，又发现了个问题：目前报的有些链接有些是 false alarms，比如 `http://127.0.0.1:2379/dashboard/#/cluster_info/instance`、`https://github.com/$user/docs-cn` 之类的链接，这些都是已知的、特意用 <> 扩起来的链接。liche 这个工具支持 exclude 掉包裹在 <> 的链接吗？还是只能用 regex 挨个过滤掉这些特定的链接呢？ @csuzhangxc

只能用 regex

**Comment by yikeke** [comment]
The `external-link-check` found two 404s. Could you help replace the two figma links? @CaitinChen 
image

**Comment by yikeke** [comment]
There are two 403s remaining that might be caused by the checker server frequently accessing the figma server:
image

I think it's okay to ignore the 403s and it's ready to merge the PR. @lilin90

File: .github/workflows/link-check.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      # NOTE:
+      #  - the check process can be very fast,
+      #    but if `-c` is too large then `Too Many Requests (HTTP error 429)` may be reported from `https://github.com/*`.
+      #  - we hardcode `--document-root` to `/github/workspace` in the container now.
+      #  - we use `http://172.*` as sample addresses in some docs, so we need to exclude them.
+      - name: Check external links
+        id: lc
+        uses: peter-evans/link-checker@v1
+        with:
+          args: -c 32 -d /github/workspace -r -x "http://172.*|https://github.com/.*/pull/.*|https://github.com/.*/issues/.*|https://github.com/[^.\n]*$|http://127.0.0.1.*|http://localhost.*|http://.*:.*|https://static.googleusercontent.com/media/research.google.com/.*|.*.md" .
```
Comment (yikeke) [inline .github/workflows/link-check.yaml:21]:
```suggestion
          args: -c 24 -d /github/workspace -r -x "http://172.*|https://github.com/.*/pull/.*|https://github.com/.*/issues/.*|https://github.com/[^.\n]*$|http://127.0.0.1.*|http://localhost.*|http://.*:.*|https://static.googleusercontent.com/media/research.google.com/.*|.*.md" .
```

**Review by CaitinChen — APPROVED** [review]

**Pull request merged**